integration-tests/src/test/scala/za/co/absa/spline/CodelessInitSpec.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
+import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructType}
+import org.apache.spark.sql.{Row, SaveMode, SparkSession}
+import org.scalatest._
+import org.spark_project.jetty.server.Server
+import org.spark_project.jetty.servlet.ServletHandler
+import za.co.absa.spline.CountingServlet._
+import za.co.absa.spline.harvester.LineageDispatcher.publishUrlProperty
+import za.co.absa.spline.harvester.SparkLineageInitializer.sparkQueryExecutionListenersKey
+import za.co.absa.spline.harvester.listener.SplineQueryExecutionListener
+
+import scala.util.Random
+
+class CodelessInitSpec extends FlatSpec with Matchers with BeforeAndAfterEach with OneInstancePerTest {
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    resetSparkContext()
+  }
+
+  override def afterEach(): Unit = {
+    resetSparkContext()
+    super.afterEach()
+  }
+
+  "codeless init" should "track lineage" in {
+    val randomPort = 8085 + new Random().nextInt(1000)
+    val session = SparkSession.builder()
+      .master("local[*]")
+      .config(sparkQueryExecutionListenersKey, classOf[SplineQueryExecutionListener].getName)
+      .config("spark." + publishUrlProperty, "http://localhost:" + randomPort)
+      .getOrCreate()
+    val schema = StructType(StructField("ID", IntegerType, nullable = false) :: StructField("NAME", StringType, nullable = false) :: Nil)
+    val rdd = session.sparkContext.parallelize(Row(1014, "Warsaw") :: Row(1002, "Corte") :: Nil)
+    val df = session.sqlContext.createDataFrame(rdd, schema)
+    serverHitsMatch(randomPort)({
+      df.write.mode(SaveMode.Overwrite).parquet("data/results/codelessInit")
+    }) shouldBe 1
+  }
+
+  /**
+    * SparkContext needs to be cleared before and after to prevent propagation of test-specific SparkConf.
+    */
+  private def resetSparkContext(): Unit = {
+    SparkSession.getActiveSession.orElse(SparkSession.getDefaultSession).foreach(_.stop())
+  }
+
+}
+
+class CountingServlet extends HttpServlet {
```

Comment:
`CodelessInitSpec.scala` file is supposed to only contain `CodelessInitSpec` class and its companion object.
`CountingServlet` class should be placed into its own file.